pulpcore/pulpcore/app/serializers/repository.py

```diff
@@ -249,6 +249,9 @@ class Meta:
             'name', 'base_path', 'publisher', 'publication', 'base_url', 'repository',
         )
 
+    def validate_base_path(self, path):
+        return self._validate_relative_path(path)
+
```

Comment:
Why do this separately from the ```validate()``` we use for everything else?

Reply:
Yes, DRF autodetects these methods:

http://www.django-rest-framework.org/api-guide/serializers/#field-level-validation

The validate method is not field specific. In this case, it's checking two fields: publisher and repository. It's better to use field-specific validation as it sets the error on the particular failing field in the response. Also, you're suppose to raise an exception which halts execution so adding this to validate would run the other validation on publisher and repository.

Reply:
Ah, I was looking at the *Django* docs, not DRF.  My bad.  And that makes sense.